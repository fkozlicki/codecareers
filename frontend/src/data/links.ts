import {
	Building2Icon,
	FilePenIcon,
	LogOutIcon,
	SettingsIcon,
	UsersIcon,
} from 'lucide-react';

export const navigationLinks = [
	{
		label: 'Applications',
		Icon: FilePenIcon,
		href: '/my-applications',
	},
	{
		label: 'Recruitments',
		Icon: UsersIcon,
		href: '/my-recruitments',
	},
	{
		label: 'Companies',
		Icon: Building2Icon,
		href: '/my-companies',
	},
	{
		label: 'Settings',
		Icon: SettingsIcon,
		href: '/settings',
	},
	{
		label: 'Logout',
		Icon: LogOutIcon,
		href: `${import.meta.env.VITE_API_URI}/logout`,
	},
];
