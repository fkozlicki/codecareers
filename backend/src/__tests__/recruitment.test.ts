import supertest from 'supertest';
import { app } from '../index.js';
import { TestDatabase, cleanupDB, initializeDB, password } from './data.js';

let jonSession = '';
let db: TestDatabase;

beforeEach(async () => {
	const database = await initializeDB();
	db = database;

	const jonResult = await supertest(app).post('/login/credentials').send({
		email: database.jon.email,
		password,
	});
	jonSession = jonResult.header['set-cookie'];
});

afterEach(async () => {
	await cleanupDB();
});

describe('Recruitment service', () => {
	describe('[GET] /recruitments', () => {
		it("Should return list of user's recruitments", async () => {
			const { statusCode, body } = await supertest(app)
				.get('/recruitments')
				.set('Cookie', jonSession);

			expect(statusCode).toBe(200);
			for (const recruitment of body.recruitments) {
				expect(recruitment).toMatchSnapshot({
					id: expect.any(String),
					jobOffer: { createdAt: expect.any(String) },
				});
			}
		});
	});

	describe('[GET] /recruitments/:id', () => {
		it('Should return recruitment', async () => {
			const { statusCode, body } = await supertest(app)
				.get(`/recruitments/${db.googleRecruitment.id}`)
				.set('Cookie', jonSession);

			expect(statusCode).toBe(200);
			expect(body.recruitment).toMatchSnapshot({
				id: expect.any(String),
				chatId: expect.any(String),
				application: {
					createdAt: expect.any(String),
					jobOffer: {
						createdAt: expect.any(String),
					},
				},
			});
		});
	});
});
