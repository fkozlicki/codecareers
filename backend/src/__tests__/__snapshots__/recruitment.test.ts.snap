// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recruitment service [GET] /recruitments Should return list of user's recruitments 1`] = `
{
  "id": Any<String>,
  "jobOffer": {
    "company": {
      "name": "Google",
    },
    "createdAt": Any<String>,
    "employmentType": "b2b",
    "level": "senior",
    "position": "Senior Web Developer",
    "salaryCurrency": "pln",
    "salaryFrom": 25000,
    "salaryTo": 30000,
    "workType": "full_time",
  },
}
`;

exports[`Recruitment service [GET] /recruitments/:id Should return recruitment 1`] = `
{
  "application": {
    "createdAt": Any<String>,
    "cv": null,
    "introduction": "Hello Google, I want to work for you",
    "jobOffer": {
      "company": {
        "avatarUrl": null,
        "backgroundUrl": null,
        "description": null,
        "name": "Google",
        "phoneNumber": "222 222 2222",
      },
      "createdAt": Any<String>,
      "description": "",
      "employmentType": "b2b",
      "level": "senior",
      "position": "Senior Web Developer",
      "salaryCurrency": "pln",
      "salaryFrom": 25000,
      "salaryTo": 30000,
      "workType": "full_time",
    },
    "user": {
      "avatar": null,
      "email": "jon.snow@gmail.com",
      "firstName": "Jon",
      "lastName": "Snow",
      "username": null,
    },
  },
  "chatId": Any<String>,
  "id": Any<String>,
}
`;
